server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -375,6 +375,47 @@ const ExpensesCollectionQuery = {
     return {
       nodes: result.rows,
       totalCount: result.count,
+      aggregation: async () => {
+        const getCurrencyTotalAmountsMemo = (() => {
```

Comment:
Is this memoization working? My intuition is that both resolvers will be called at the same time, calling the function twice in parallel - unless we're lucky with the timing.

The proper way to achieve that would normally be to implement a loader.

Reply:
Yes, thanks! There was an issue there. In this last push, I refactored to use a promise. It will only execute once and memoize the result.

I think a loader would be a bit more complex due to passing the collection query filters to it. And since this is not used in a list of expenses lists, maybe not needed.

Reply:
Followed up in https://github.com/opencollective/opencollective-api/pull/9456/files#r1399291105